backend/src/test/java/org/eclipse/tractusx/puris/backend/erpadapter/logic/service/ErpAdapterRequestClientTest.java

```diff
@@ -0,0 +1,130 @@
+
+    private static final String supplierPartnerBpnl = "BPNL1234567890ZZ";
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final String apiKey = "x-api-key";
+
+    private static final String apiSecret = "my-secret";
+
+    private static final String requestType = "itemstock";
+
+    private static final String sammVersion = "2.0";
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        mockWebServer = new MockWebServer();
+        mockWebServer.start();
+        erpAdapterRequestClient = new ErpAdapterRequestClient(variablesService);
+
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        mockWebServer.shutdown();
+    }
+
+    private void prepareVariablesService() {
+        Mockito.when(variablesService.getErpAdapterUrl()).thenReturn(mockWebServer.url("/").toString());
+        Mockito.when(variablesService.getErpAdapterAuthKey()).thenReturn(apiKey);
+        Mockito.when(variablesService.getErpAdapterAuthSecret()).thenReturn(apiSecret);
+        Mockito.when(variablesService.getErpResponseUrl()).thenReturn(erpResponseUrl);
+    }
+
+    @Test
+    public void test_should_success() throws Exception {
+        // given
+        UUID uuid = UUID.randomUUID();
+        ErpAdapterRequest erpAdapterRequest = ErpAdapterRequest.builder()
+            .requestDate(new Date())
+            .partnerBpnl(supplierPartnerBpnl)
+            .id(uuid)
+            .directionCharacteristic(DirectionCharacteristic.INBOUND)
+            .ownMaterialNumber(matNbrCustomer)
+            .requestType(requestType)
+            .sammVersion(sammVersion)
+            .build();
+
+        // when
+        prepareVariablesService();
+        erpAdapterRequestClient.sendRequest(erpAdapterRequest);
+        RecordedRequest request = mockWebServer.takeRequest();
+
+        // then
+        Assertions.assertThat(request.getMethod()).isEqualTo("POST");
+
+        Assertions.assertThat(request.getHeader(apiKey)).isEqualTo(apiSecret);
+        Assertions.assertThat(request.getHeader("Content-type")).contains("application/json");
+
+        Assertions.assertThat(request.getPath()).contains(supplierPartnerBpnl);
+        Assertions.assertThat(request.getPath()).contains(requestType);
+        Assertions.assertThat(request.getPath()).contains(sammVersion);
+        Assertions.assertThat(request.getPath()).contains(uuid.toString());
+
+        try (InputStream stream = request.getBody().inputStream()) {
+            JsonNode requestBodyNode = objectMapper.readTree(new String(stream.readAllBytes()));
+            Assertions.assertThat(requestBodyNode.get("material").asText()).isEqualTo(matNbrCustomer);
+            Assertions.assertThat(requestBodyNode.get("direction").asText()).isEqualTo(DirectionCharacteristic.INBOUND.toString());
+            Assertions.assertThat(requestBodyNode.get("response-url").asText()).isEqualTo(erpResponseUrl);
+        }
+
```

Comment:
please remove duplicate blank line and add a trailing blank line at end of file :)